lib/Dialect/Triton/IR/Ops.cpp

```diff
@@ -704,18 +706,19 @@ LogicalResult ReshapeOp::verify() {
 
   if (srcEnc && !getAllowReorder()) {
     Attribute inferredDstEnc;
-    if (cast<DialectInferLayoutInterface>(&srcEnc.getDialect())
-            ->inferReshapeOpNoReorderEncoding(srcTy.getShape(), srcEnc,
-                                              dstTy.getShape(), inferredDstEnc,
-                                              getLoc())
-            .failed()) {
-      return emitError("This reshape is impossible without reordering, but "
-                       "reordering is not allowed.  Try choosing a different "
-                       "encoding for the input tensor (or allow reordering).");
-    }
+    assert(succeeded(cast<DialectInferLayoutInterface>(&srcEnc.getDialect())
+                         ->inferReshapeOpEncoding(srcTy.getShape(), srcEnc,
+                                                  dstTy.getShape(),
+                                                  inferredDstEnc, getLoc())));
```

Comment:
```suggestion
    [[maybe_unused]] auto result = cast<DialectInferLayoutInterface>(&srcEnc.getDialect())
                         ->inferReshapeOpEncoding(srcTy.getShape(), srcEnc,
                                                  dstTy.getShape(),
                                                  inferredDstEnc, getLoc());
    assert(succeeded(result));
```

Asserts should not be side-effecting, otherwise this code will break when asserts are off.

Reply:
ooops, thank you for catching this one!